### ISSUE TITLE
Issue in Serenity v3.3.1 with automatically creating chrome driver capabilities through Serenity.conf 

### ISSUE THREAD

**Issue by sandeepraina-lenze** [serenity-bdd/serenity-core#2888]
Hi,

I found an issue in **Serenity v3.3.1** where chrome driver capabilities creating through _Serenity.conf_ is not happening properly

If the _Serenity.conf_ is created as per **W3C** standards

`webdriver {
    driver = "${driver}"
    autodownload = true
    timeouts {
        implicitlywait = 15000
        fluentwait = 15000
    }
   capabilities {
    browserName = "Chrome"
    "goog:chromeOptions" {
      args = [ "--window-size=1000,800", "--headless" ]
    }
  }
}`

Then the chrome driver capabilities created are as follow

`17:30:01.926 [main] DEBUG net.serenitybdd.core.webdriver.driverproviders.ProvideNewDriver - Instantiating driver
17:30:01.926 [main] DEBUG net.serenitybdd.core.webdriver.driverproviders.ProvideNewDriver - Driver capabilities: Capabilities {acceptInsecureCerts: false, autodownload: true, browserName: chrome, capabilities."goog:chromeOptions".args: [--window-size=1000,800, --headless], capabilities.browserName: Chrome, driver: chrome, goog:chromeOptions: {args: [], extensions: []}, timeouts.fluentwait: 15000, timeouts.implicitlywait: 15000}`

This configuration will not add any value and no affect is seen.

Whereas if the configuration is created using _chrome.switches_ 

`webdriver {
    driver = "${driver}"
    autodownload = true
    timeouts {
        implicitlywait = 15000
        fluentwait = 15000
    }
}
chrome.switches = """--headless;--start-maximized;--test-type;--no-sandbox;--ignore-certificate-errors;--disable-popup-blocking;--disable-default-apps;--disable-extensions-file-access-check;--disable-infobars;--disable-gpu;--user-agent=Apache;--disable-dev-shm-usage;--window-size=1920,1080"""` 

Then configuration will take affect as chrome capabilities get created properly

`17:33:17.564 [main] DEBUG net.serenitybdd.core.webdriver.driverproviders.ProvideNewDriver - Instantiating driver
17:33:17.564 [main] DEBUG net.serenitybdd.core.webdriver.driverproviders.ProvideNewDriver - Driver capabilities: Capabilities {acceptInsecureCerts: false, autodownload: true, browserName: chrome, driver: chrome, goog:chromeOptions: {args: [--headless, --start-maximized, --test-type, --no-sandbox, --ignore-certificate-errors, --disable-popup-blocking, --disable-default-apps, --disable-extensions-file-a..., --disable-infobars, --disable-gpu, --user-agent=Apache, --disable-dev-shm-usage, --window-size=1920,1080], extensions: []}, timeouts.fluentwait: 15000, timeouts.implicitlywait: 15000}`

Kindly look into this issue and let me know if more details/information required from my end.

**Comment by wakaleo** [comment]
`chrome.switches` is deprecated in 3.3.x. Maybe have a look in the ChromiumOptionsBuilder class to see if you can find anything and propose a fix?

**Comment by wakaleo** [comment]
Also, make sure browsername is not capitalised, e.g.
```
webdriver {
  driver = chrome
  autodownload = true
  capabilities {
    browserName = "chrome"
    acceptInsecureCerts = true
    "goog:chromeOptions" {
      args = ["test-type", "ignore-certificate-errors",  "--window-size=1000,800"
        "incognito", "disable-infobars", "disable-gpu", "disable-default-apps", "disable-popup-blocking"]
    }
  }
}
```

**Comment by sandeepraina-lenze** [comment]
Thank you for quick update! But I am still able to reproduce the issue.

11:05:03.219 [main] DEBUG net.serenitybdd.core.webdriver.driverproviders.ProvideNewDriver - Instantiating driver
11:05:03.219 [main] DEBUG net.serenitybdd.core.webdriver.driverproviders.ProvideNewDriver - Driver capabilities: Capabilities {acceptInsecureCerts: false, autodownload: true, browserName: chrome, capabilities."goog:chromeOptions".args: [test-type, ignore-certificate-errors, --window-size=1000,800, incognito, disable-infobars, disable-gpu, disable-default-apps, disable-popup-blocking], capabilities.acceptInsecureCerts: true, capabilities.browserName: chrome, driver: chrome, goog:chromeOptions: {args: [], extensions: []}}

Below is my entire **Serenity.conf** file, any sugestion

`webdriver {
  driver = chrome
  autodownload = true
  capabilities {
    browserName = "chrome"
    acceptInsecureCerts = true
    "goog:chromeOptions" {
      args = ["test-type", "ignore-certificate-errors",  "--window-size=1000,800"
        "incognito", "disable-infobars", "disable-gpu", "disable-default-apps", "disable-popup-blocking"]
    }
  }
}

serenity {
  take.screenshots = FOR_EACH_ACTION

  project.name = "${reporttitle}"
  compress.filenames = false
  
  browser {
    maximized = true
  }

  report {
    tagtypes="capability,feature"
    hide.empty.requirements = true
    exclude.tags="startrecording,staging,reset,stoprecording,notdeployed,debug,singlebrowser"
  }
}

environments {
  with_screenshots {
    serenity {
      take.screenshots = BEFORE_AND_AFTER_EACH_STEP
    }
  }
}

shutterbug {
    capturestrategy = FULL
    betweenScrollTimeout = 100
    useDevicePixelRatio = true
}`

**Comment by wakaleo** [comment]
That looks like it should work - did you check for dependency conflicts?

**Comment by sandeepraina-lenze** [comment]
I do not think so sharing my pom file for reference

```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>org.test.automation</groupId>
    <artifactId>automation-qa</artifactId>
    <version>1.0.0</version>
    <packaging>jar</packaging>

    <name>Test Automation Framework</name>

    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <serenity.core.version>3.3.1</serenity.core.version>
        <serenity.cucumber.version>3.3.1</serenity.cucumber.version>
        <assertJ.version>3.23.1</assertJ.version>
        <logback.version>1.2.11</logback.version>
        <unirest.version>3.13.10</unirest.version>
        <ffmpeg.version>0.7.0</ffmpeg.version>
        <gson.version>2.9.1</gson.version>
        <maven.plugins.version>3.0.0-M7</maven.plugins.version>
        <maven.compiler.plugin.version>3.10.1</maven.compiler.plugin.version>

        <tags></tags>
        <driver></driver>
        <reporttitle></reporttitle>
        <proxyHost></proxyHost>
        <proxyPort></proxyPort>

        <encoding>UTF-8</encoding>
    </properties>
    <dependencies>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-core</artifactId>
            <version>${serenity.core.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-cucumber</artifactId>
            <version>${serenity.cucumber.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-screenplay</artifactId>
            <version>${serenity.core.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-screenplay-webdriver</artifactId>
            <version>${serenity.core.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-screenplay-rest</artifactId>
            <version>${serenity.core.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-ensure</artifactId>
            <version>${serenity.core.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-shutterbug1x</artifactId>
            <version>${serenity.core.version}</version>
        </dependency>
        <dependency>
            <groupId>org.junit.platform</groupId>
            <artifactId>junit-platform-launcher</artifactId>
            <version>1.9.0</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.junit.vintage</groupId>
            <artifactId>junit-vintage-engine</artifactId>
            <version>5.9.0</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.junit.jupiter</groupId>
            <artifactId>junit-jupiter-engine</artifactId>
            <version>5.9.0</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>ch.qos.logback</groupId>
            <artifactId>logback-classic</artifactId>
            <version>${logback.version}</version>
        </dependency>
        <dependency>
            <groupId>com.konghq</groupId>
            <artifactId>unirest-java</artifactId>
            <version>${unirest.version}</version>
        </dependency>
        <dependency>
            <groupId>net.bramp.ffmpeg</groupId>
            <artifactId>ffmpeg</artifactId>
            <version>${ffmpeg.version}</version>
        </dependency>
        <dependency>
            <groupId>com.google.code.gson</groupId>
            <artifactId>gson</artifactId>
            <version>${gson.version}</version>
        </dependency>
        <dependency>
            <groupId>org.assertj</groupId>
            <artifactId>assertj-core</artifactId>
            <version>${assertJ.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-single-page-report</artifactId>
            <version>${serenity.core.version}</version>
        </dependency>
    </dependencies>
    <build>
        <plugins>
            <plugin>
                <groupId>net.serenity-bdd.maven.plugins</groupId>
                <artifactId>serenity-maven-plugin</artifactId>
                <version>${serenity.core.version}</version>
                <dependencies>
                    <dependency>
                        <groupId>net.serenity-bdd</groupId>
                        <artifactId>serenity-single-page-report</artifactId>
                        <version>${serenity.core.version}</version>
                    </dependency>
                </dependencies>
                <configuration>
                    <reports>single-page-html</reports>
                </configuration>
                <executions>
                    <execution>
                        <id>serenity-reports</id>
                        <phase>post-integration-test</phase>
                        <goals>
                            <goal>aggregate</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>${maven.plugins.version}</version>
                <configuration>
                    <skip>true</skip>
                </configuration>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-failsafe-plugin</artifactId>
                <version>${maven.plugins.version}</version>
                <configuration>
                    <includes>
                        <include>**/*TestSuite.java</include>
                    </includes>
                    <systemPropertyVariables>
                        <context>${driver}</context>
                        <injected.tags>${driver}</injected.tags>
                        <https.proxyHost>${proxyHost}</https.proxyHost>
                        <https.proxyPort>${proxyPort}</https.proxyPort>
                        <http.proxyHost>${proxyHost}</http.proxyHost>
                        <http.proxyPort>${proxyPort}</http.proxyPort>
                    </systemPropertyVariables>
                    <parallel>none</parallel>
                    <useUnlimitedThreads>false</useUnlimitedThreads>
                </configuration>
                <executions>
                    <execution>
                        <goals>
                            <goal>integration-test</goal>
                            <goal>verify</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>${maven.compiler.plugin.version}</version>
                <configuration>
                    <source>17</source>
                    <target>17</target>
                </configuration>
            </plugin>
        </plugins>
    </build>
</project>
```

**Comment by wakaleo** [comment]
I can't reproduce it, the fastest thing would be for you to dig into the driver initiation code to see what it is doing.

**Comment by sandeepraina-lenze** [comment]
Thank you John, Let me check this and share the feedback with you

**Comment by sandeepraina-lenze** [comment]
Hi John, I guess I found out the reason of this issue after doing a dry run of your code base. In my case always a chrome driver configuration is happening through **ProvideNewDriver.java** instead of **ChromeDriverProvider.java**. So need your guidance I initialize driver through maven command like this _mvn verify -Ddriver=chrome_ Is this right as I think it is not able to recognize the correct driver to be initialized and is using default method

**Comment by wakaleo** [comment]
Hmmm, the ProvideNewDriver class is no longer used in 3.3.1 - do you have a stack trace showing where it is being called from?

**Comment by sandeepraina-lenze** [comment]
Hi John, I glad to tell you that the issue is resolved. When I was about to send you the stack trace, I realized that we have created a model which runs on both Gradle and Maven. So I made the changes in the **pom** file but did not made the changes in the **build.gradle** so it continues to use the old references. Hence, when I triggered my test through _intellij_ it was by default triggering through _gradle_. After fixing this every thing worked perfectly well. Thank you for looking into this.
